chsdi/lib/opentransapi/opentransapi.py

```diff
@@ -0,0 +1,95 @@
+            'label': el.find('./trias:Service/trias:PublishedLineName/trias:Text', ns).text,
+            'currentDate': self._format_time(root.find('.//{http://www.siri.org.uk/siri}ResponseTimestamp').text),
+            'departureDate': self._format_time(el.find('./trias:ThisCall/trias:CallAtStop/trias:ServiceDeparture/trias:TimetabledTime', ns).text),
+            'estimatedDate': self._convert_estimated_date(el.find('./trias:ThisCall/trias:CallAtStop/trias:ServiceDeparture/trias:EstimatedTime', ns)),
+            'destinationName': el.find('./trias:Service/trias:DestinationText/trias:Text', ns).text,
+            'destinationId': el.find('./trias:Service/trias:DestinationStopPointRef', ns).text
+        } for el in el_stop_events]
+        if el_stop_events == None or results == []:
+            results = [{'destination': 'nodata'}]
+        return results
+
+    def send_post(self, station_id, request_dt_time, number_results=5):
+        self.headers = {
+            'authorization': self.open_trans_api_key,
+            'content-type': 'application/xml; charset=utf-8',
+            'accept-charset': 'utf-8'
+        }
+        self.xml_data = u"""
+                    <?xml version="1.0" encoding="UTF-8"?>
+                    <Trias version="1.1" xmlns="http://www.vdv.de/trias" xmlns:siri="http://www.siri.org.uk/siri" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+                        <ServiceRequest>
+                            <siri:RequestTimestamp>%s</siri:RequestTimestamp>
+                            <siri:RequestorRef>EPSa</siri:RequestorRef>
+                            <RequestPayload>
+                                <StopEventRequest>
+                                    <Location>
+                                        <LocationRef>
+                                            <StopPointRef>%s</StopPointRef>
+                                        </LocationRef>
+                                        <DepArrTime>%s</DepArrTime>
+                                    </Location>
+                                    <Params>
+                                        <NumberOfResults>%s</NumberOfResults>
+                                        <StopEventType>departure</StopEventType>
+                                        <IncludePreviousCalls>false</IncludePreviousCalls>
+                                        <IncludeOnwardCalls>true</IncludeOnwardCalls>
+                                        <IncludeRealtimeData>true</IncludeRealtimeData>
+                                    </Params>
+                                </StopEventRequest>
+                            </RequestPayload>
+                        </ServiceRequest>
+                    </Trias>
+                    """ % (str(request_dt_time), str(station_id), str(request_dt_time), str(number_results))
+
+        self.r = requests.post(self.url, self.xml_data, headers=self.headers, timeout=5)
```

Comment:
Hi you may want to use ConnectionError, in case DNS resolution doesn't work or simply because their server is not responding.

https://www.programcreek.com/python/example/50847/requests.exceptions.ConnectionError